### ISSUE TITLE
Multiple instances of `Undefined behavior: left shift of negative value`

### ISSUE THREAD

**Issue by pankajp** [enthought/enable#221]
Compiling enable with UBSan `-fsanitize=undefined` http://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html spews the following messages when running the demo app and cycling through the demos.

```
$ python examples/demo/demo.py 
loaded the Generic plugin 
kiva/agg/freetype2/src/base/ftstream.c:413:22: runtime error: left shift of negative value -114
kiva/agg/freetype2/src/sfnt/ttmtx.c:195:24: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/sfnt/ttmtx.c:207:16: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/base/ftstream.c:357:22: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/sfnt/ttcmap.c:900:18: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/truetype/ttgload.c:310:25: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/truetype/ttgload.c:492:21: runtime error: left shift of negative value -2
kiva/agg/freetype2/src/truetype/ttgload.c:528:21: runtime error: left shift of negative value -2
kiva/agg/freetype2/src/truetype/ttgload.c:309:25: runtime error: left shift of negative value -1
kiva/agg/agg-24/include/agg_rasterizer_scanline_aa.h:262:56: runtime error: left shift of negative value -61
kiva/agg/agg-24/include/agg_rasterizer_scanline_aa.h:272:55: runtime error: left shift of negative value -256
kiva/agg/freetype2/src/sfnt/ttkern.c:250:25: runtime error: left shift of negative value -1
kiva/agg/agg-24/include/agg_path_storage_integer.h:45:20: runtime error: left shift of negative value -63
kiva/agg/agg-24/include/agg_path_storage_integer.h:44:20: runtime error: left shift of negative value -5
kiva/agg/agg-24/include/agg_rasterizer_scanline_aa.h:262:56: runtime error: left shift of negative value -128
kiva/agg/agg-24/include/agg_rasterizer_scanline_aa.h:272:55: runtime error: left shift of negative value -128
kiva/agg/freetype2/src/truetype/ttgload.c:312:25: runtime error: left shift of negative value -2
kiva/agg/agg-24/include/agg_rasterizer_cells_aa.h:367:36: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/sfnt/ttcmap.c:1130:18: runtime error: left shift of negative value -1
kiva/agg/freetype2/src/sfnt/ttcmap.c:690:25: runtime error: left shift of negative value -1
```

**Comment by corranwebster** [comment]
These are in vendorized libraries (freetype, agg) which we've just included wholesale in the package.  The libraries are probably getting on for a decade old at this point, but we're limited in ability to update: agg because of licensing issues (it went GPL) and not sure about freetype (but its work to do it).

So the correct fix would be to update those two packages as well as we can within the constraints of licenses - simplest would be to look at what matplotlib uses currently.

**Comment by pankajp** [comment]
I suspected some of these undefined behaviors might be cause of segfault ( https://github.com/enthought/enable/issues/97#issuecomment-28299604 ) due to over-optimistic compiler optimizations in presence of undefined behavior, which is why i reported them here.

**Comment by corranwebster** [comment]
Not saying that they aren't a potential problem, just emphasising that updating agg and freetype is probably a better fix than patching what we currently have (assuming that there are fixes upstream).

**Comment by pankajp** [comment]
Yes, that definitely makes sense. I'll see what matplotlib is upto, and whether directly using mpl's bundled agg might work for us. Thanks

**Comment by pankajp** [comment]
And i suspect you might be correct: Matplotlib bundles https://sourceforge.net/p/agg/svn/94/ "Changed parameter types from const to non-const reference, to prevent dangling references to temporary variables" which might be the fix we need

**Comment by jwiggins** [comment]
*cough* *cough* https://github.com/jwiggins/pyagg *cough* *cough*

:smirk:

**Comment by jdeschenes** [comment]
I think this issue may be closed with #288.

**Comment by jwiggins** [comment]
Closing
